### PR TITLE
Utterances now stitched if {stitch: true} part of a find request.

### PR DESCRIPTION
**Pull request by dcalacci**
- Default parameters are that utterances under 5s apart are stitched together,
  and any utterances under .5s are removed from results.

## Description
added a simple after hook to stitch utterances together and remove small ones

## Motivation and context

motivation here is to provide more human-friendly data from the recorded utterances

## How has this been tested?
tested visually by having multiple meetings with different participants over global network and then inspecting timeline data on the riff-rtc dashboard. 

## Types of changes

- ✅ Bug fix (non-breaking change which fixes an issue)
- ✅ New feature (non-breaking change which adds functionality)
- ✅ Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ My change requires a change to the documentation and I have updated it accordingly.
- ✅ I have added tests to cover my changes.

### CODE REVIEW

**Comment by dcalacci** [comment]
[Influence graph is not easy to understand. Students/novices don't know what that graph represents or means](https://trello.com/c/BNV1gyGt/150-influence-graph-is-not-easy-to-understand-students-novices-dont-know-what-that-graph-represents-or-means)

**Comment by dcalacci** [comment]
@jordanreedie could you just quickly review this? @mlippert once jord OKs it can you merge and push to staging?

File: src/services/utterance/hooks/stitchHook.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            utts[i].endTime = utts[i+1].endTime;
+            utts.splice(i+1,1); // remove merged utterance
+        }
+    }
+
+    winston.log('info', "iterating over ", utts.length, 'utterances.')
+    var longerUtts = []
+    for (var j = 0; j < utts.length; j++) {
+        // if this utterance is under .5s
+        if ((Math.abs(utts[j].endTime - utts[j].startTime)) > 1000) {
```
Comment (jordanreedie) [inline src/services/utterance/hooks/stitchHook.js]: wait isn't this checking if the utterance is under 1 second long or am i being a dummy

File: src/services/utterance/hooks/stitchHook.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    if (utts.length === 0) {
+        // nothing to see here
+        return hook
+    }
+
+    winston.log('info', "iterating over ", utts.length, 'utterances.')
+
+    for (var i = 0; i < utts.length-1; i++) {
+        // if this utterance ended within 2s of the next start
+        if ((Math.abs(utts[i].endTime - utts[i+1].startTime)) < 5000) {
```
Comment (jordanreedie) [inline src/services/utterance/hooks/stitchHook.js]: the comment here just needs an update

File: src/services/utterance/hooks/stitchHook.js (hunk 1 above)
Comment (jordanreedie) [reply src/services/utterance/hooks/stitchHook.js]: oops i didn't make this a review comment

**Comment by dcalacci** [comment]
@jordanreedie should be fixed in latest commit

**Pull request merged**